### PR TITLE
Add Admin Functionality

### PR DESCRIPTION
**Pull request by aditi-webonise**
- Add rooms
- Delete rooms
- Update Rooms
- Book room

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,82 +1,77 @@
...
+        <dependency>
+            <groupId>net.sourceforge.nekohtml</groupId>
+            <artifactId>nekohtml</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
```
Comment (pareekshita-webonise) [inline pom.xml]: why same dependency repeated?

File: pom.xml [hunk 2]
```diff
@@ -1,82 +1,77 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>net.sourceforge.nekohtml</groupId>
+            <artifactId>nekohtml</artifactId>
```
Comment (pareekshita-webonise) [inline pom.xml]: what is the use of this dependency?

File: pom.xml [hunk 3]
```diff
@@ -1,82 +1,77 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
```
Comment (pareekshita-webonise) [inline pom.xml]: again what is the use?

File: src/main/java/com/webonise/rbs/controller/IndexController.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+@Controller
+public class IndexController {
+
+    @GetMapping(value = "/")
+    public ModelAndView index() {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName("index");
+        return modelAndView;
+    }
+}
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/controller/IndexController.java]: add a blank line to EOF.

File: src/main/java/com/webonise/rbs/controller/RoomController.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+@RequestMapping(value = "/room")
+public class RoomController {
+
+    @Autowired
+    RoomServiceImpl roomService;
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/controller/RoomController.java]: make it private and you have to autowire the interface not the impl class.

File: src/main/java/com/webonise/rbs/entity/Booking.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
+package com.webonise.rbs.entity;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.sql.Timestamp;
+
+@Entity
+@Table(name="bookings")
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/entity/Booking.java]: Table name should be singular.

File: src/main/java/com/webonise/rbs/controller/RoomController.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                model.addAttribute("editRoom", editRoom);
+                return "editRoom";
+            } else {
+                redirectAttributes.addFlashAttribute("status","notfound");
+            }
+        }
+        return "redirect:/room/";
+    }
+
+    @PostMapping(value = "/update")
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/controller/RoomController.java]: for update PUT method is used by convention.

File: src/main/java/com/webonise/rbs/controller/RoomController.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        if(roomService.addRoom(room)!=null) {
+            redirectAttributes.addFlashAttribute("saveRoom", "success");
+        } else {
+            redirectAttributes.addFlashAttribute("saveRoom", "failure");
+        }
+        return "redirect:/room/";
+    }
+
+    @GetMapping(value = "/{operation}/{id}")
+    public String editRemoveRoom(@PathVariable("operation") String operation, @PathVariable("id") Long id, final RedirectAttributes redirectAttributes, Model model) {
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/controller/RoomController.java]: How is deleting a room controller a GET mapping?

File: src/main/java/com/webonise/rbs/entity/Role.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "roles")
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/entity/Role.java]: singular table name

File: src/main/java/com/webonise/rbs/entity/Booking.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+public class Booking implements Serializable {
+
+    private static final long serialVersionUID = -843414259970947020L;
+
+    @Id
+    private Long id;
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/entity/Booking.java]: Generation Type??

File: src/main/java/com/webonise/rbs/service/BookingServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class BookingServiceImpl implements BookingService {
+
+    @Autowired
+    BookingRepository bookingRepository;
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/service/BookingServiceImpl.java]: make it private

File: src/test/java/com/webonise/rbs/RbsApplicationTest.java [hunk 12]
```diff
@@ -1,3 +1,4 @@
+/*
 package com.webonise.rbs;
 
 import org.junit.Test;
```
Comment (pareekshita-webonise) [inline src/test/java/com/webonise/rbs/RbsApplicationTest.java]: do not push commented code. If it is not required then remove it.

File: src/main/java/com/webonise/rbs/controller/RoomController.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    RoomServiceImpl roomService;
+
+    @GetMapping(value = "/",produces = MediaType.TEXT_HTML_VALUE)
+    public String savePage(Model model) {
+        model.addAttribute("room", new Room());
+        model.addAttribute("allRooms", roomService.getAllRooms());
+        return "viewRooms";
+    }
+
+    @PostMapping(value = "/save")
```
Comment (akshaym-webonise) [inline src/main/java/com/webonise/rbs/controller/RoomController.java]: Please follow Rest APIs conventions 
http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api

File: src/main/java/com/webonise/rbs/controller/IndexController.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
+package com.webonise.rbs.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.servlet.ModelAndView;
+
+@Controller
+public class IndexController {
+
+    @GetMapping(value = "/")
```
Comment (pareekshita-webonise) [inline src/main/java/com/webonise/rbs/controller/IndexController.java]: no need to specify "/". If you specify then you will get data on "www.abc/" not "www.abc"

File: src/main/java/com/webonise/rbs/controller/RoomController.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+            redirectAttributes.addFlashAttribute("saveRoom", "success");
+        } else {
+            redirectAttributes.addFlashAttribute("saveRoom", "failure");
+        }
+        return "redirect:/room/";
+    }
+
+    @GetMapping(value = "/{operation}/{id}")
+    public String editRemoveRoom(@PathVariable("operation") String operation, @PathVariable("id") Long id, final RedirectAttributes redirectAttributes, Model model) {
+        if(operation.equals("delete")) {
```
Comment (akshaym-webonise) [inline src/main/java/com/webonise/rbs/controller/RoomController.java]: What does this method do actually? It seems that it tries to do 2 task according to operation type. Not recommended.

**Review by pareekshita-webonise — CHANGES_REQUESTED** [review]

**Review by pareekshita-webonise — APPROVED** [review]

**Pull request merged**